### PR TITLE
docs: add casdoor as identity provider

### PR DESCRIPTION
**Pull request by Abingcbc**
## Description
[Casdoor](https://github.com/casdoor/casdoor) is a UI-first centralized authentication / Single-Sign-On (SSO) platform supporting OAuth 2.0, OIDC and SAML, integrated with Casbin RBAC and ABAC permission management.

## Motivation and Context
We want to support MinIO's SSO login by Casdoor. Therefore, we updated documents for users to configure Casdoor as an identity provider for MinIO.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: docs/sts/README.md [hunk 1]
```diff
@@ -49,6 +50,15 @@ export MINIO_IDENTITY_OPENID_CLIENT_ID="843351d4-1080-11ea-aa20-271ecba3924a"
 minio server /mnt/data
 ```
 
+#### Casdoor
+```
+export MINIO_ROOT_USER=minio
+export MINIO_ROOT_PASSWORD=minio123
+export MINIO_IDENTITY_OPENID_CONFIG_URL=http://localhost:8000/.well-known/openid-configuration
```
Comment (klauspost) [inline docs/sts/README.md]: In your other doc, you have:
```suggestion
export MINIO_IDENTITY_OPENID_CONFIG_URL=http://`CASDOOR_HOSTNAME`:8000/.well-known/openid-configuration
```
Or maybe
```suggestion
export MINIO_IDENTITY_OPENID_CONFIG_URL=http://CASDOOR_HOSTNAME:CASDOOR_PORT/.well-known/openid-configuration
```

File: docs/sts/casdoor.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    - Copy `Client ID` and `Client secret`
+    - Add your redirect url (callback url) to `Redirect URLs`
+    - Save
+
+- Go to Users
+  - Edit the user
+    - Add your MinIO policy (ex: `readwrite`) in `Tag`
+    - Save
+
+- Open your favorite browser and visit: **http://`CASDOOR_HOSTNAME`/.well-known/openid-configuration**, you will see the OIDC configure of Casdoor.
```
Comment (klauspost) [inline docs/sts/casdoor.md]:
```suggestion
- Open your favorite browser and visit: **http://`CASDOOR_HOSTNAME`:`CASDOOR_PORT`/.well-known/openid-configuration**, you will see the OIDC configure of Casdoor.
```

File: docs/sts/casdoor.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+MINIO_IDENTITY_OPENID_CLIENT_ID     (string)    unique public identifier for apps e.g. "292085223830.apps.googleusercontent.com"
+MINIO_IDENTITY_OPENID_CLAIM_NAME    (string)    JWT canned policy claim name, defaults to "policy"
+MINIO_IDENTITY_OPENID_CLAIM_PREFIX  (string)    JWT claim namespace prefix e.g. "customer1/"
+MINIO_IDENTITY_OPENID_SCOPES        (csv)       Comma separated list of OpenID scopes for server, defaults to advertised scopes from discovery document e.g. "email,admin"
+MINIO_IDENTITY_OPENID_COMMENT       (sentence)  optionally add a comment to this setting
+```
+
+Set `identity_openid` config with `config_url`, `client_id` and restart MinIO
+```
+~ mc admin config set myminio identity_openid config_url="http://`CASDOOR_HOSTNAME`/.well-known/openid-configuration" client_id=<client id> client_secret=<client secret> claim_name="tag"
```
Comment (klauspost) [inline docs/sts/casdoor.md]:
```suggestion
~ mc admin config set myminio identity_openid config_url="http://CASDOOR_HOSTNAME:CASDOOR_PORT/.well-known/openid-configuration" client_id=<client id> client_secret=<client secret> claim_name="tag"
```

**Review by klauspost — COMMENTED** [review]
Thanks for the contribution, and for supporting MinIO directly.

I don't know the configuration, so maybe "port" is usually just the default. Adjust as you see fit.

**Comment by Abingcbc** [comment]
> Thanks for the contribution, and for supporting MinIO directly.
> 
> I don't know the configuration, so maybe "port" is usually just the default. Adjust as you see fit.

Thanks for your reply. I have fixed all according to your suggestion :)

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by Abingcbc** [comment]
> Thanks for the contribution, and for supporting MinIO directly.
> 
> I don't know the configuration, so maybe "port" is usually just the default. Adjust as you see fit.

Maybe `hostname` is confusing. So I use `CASDOOR_ENDPOINT` to replace `CASDOOR_HOSTNAME:CASDOOR_PORT`.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by klauspost — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**